modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -33,37 +31,48 @@ export class BaseAPI {
         return this.withMiddleware<T>(...middlewares);
     }
 
-    protected async request(context: RequestOpts, initOverrides?: RequestInit): Promise<Response> {
-        const { url, init } = this.createFetchParams(context, initOverrides);
+    protected async request(context: RequestOpts, initOverrides?: RequestInit | ((requestContext: { init: HTTPRequestInit, context: RequestOpts }) => Promise<RequestInit>)): Promise<Response> {
```

Comment:
I feel that the initOverrides async function's interface is too flexible.

For use cases like the [original](https://github.com/OpenAPITools/openapi-generator/issues/11823) ISSUE, perhaps context should be dropped from initial support.

The reason context is given as an argument in this function is because it was given as a sample interface for [original](https://github.com/OpenAPITools/openapi-generator/issues/11823) ISSUE.
However, I believe that passing context is out of the use case of the ISSUE.